Steering Council DEP (#75)
django/deps: draft/0012-steering-council.rst

==============================
DEP 0012: The Steering Council
==============================

:DEP: 0012
:Author: Andrew Godwin
:Implementation Team: Andrew Godwin

> :Shepherd: TBD

Reviewer: I'm happy to act as Shepherd if you're in need of one Andrew.

Author: Thanks!

:Status: Draft
:Type: Process
:Created: 2022-10-26
:Last-Modified: 2022-10-26

.. contents:: Table of Contents
   :depth: 3
   :local:

Abstract
========

> The Technical Board, as defined in DEP 10, will be renamed to the Steering

Reviewer 2: ```suggestion
The Technical Board, as defined in `DEP 10 <https://github.com/django/deps/blob/main/accepted/0010-new-governance.rst>`_, will be renamed to the Steering
```

Council, the eligibility requirements for becoming a candidate will be
revised, and a renewed focus will be placed on inviting feature changes.

Specification
=============

The Technical Board, as specified in DEP 10, will be renamed to the Steering
Council. All other aspects of DEP 10 will continue to apply unless noted below.

The existing requirements to be qualified for election to the Steering Council
will be replaced with requiring both of the following:

* A history of substantive contributions to Django or the Django
  ecosystem. This history must begin at least 18 months prior to the
  individual's candidacy for the Steering Council, and include substantive
  contributions in at least two of these bullet points:

  * Code contributions on Django projects or major third-party packages in
    the Django ecosystem

  * Reviewing pull requests and/or triaging Django project tickets

  * Documentation, tutorials or blog posts

>   * Discussions about Django on the django-developers mailing list or the

Reviewer 3: Is the omission of other "official" media like IRC or the Discord deliberate or accidental?

What about other places like SO?

Author: This is as much of a diff as I can reasonably write, in that the DEP is basically a diff onto DEP 10 :)

If you're curious, the only additions here are documentation and events/user groups, and then allowing people to have some combination rather than requiring the existing three.

    Django Forum

  * Running Django-related events or user groups

* A history of recent engagement with the direction and future of
  Django. Such engagement must have occurred within a period of no
  more than two years prior to the individual's candidacy for the

>   Steering Council.

Reviewer: This ≈ comes from the existing text right? 

I understand the motivation, but just want to query if we **do** want to keep it? The discussion on the DSF members list had a post from @frankwiles (amongst others) who is clearly qualified to serve on the Steering Council but (paraphrasing his words) hasn't been active recently, and so may be excluded. 

I don't 100% know that's the conclusion, but that's the fear

Do we want to allow people who've been rotated out, for whatever life reasons, but at a given election think "yes, I want to get back in"? 

I think perhaps we do. 

Other long-standing members of the community who have been inactive said at DjangoCon that they were considering getting back involved. It would be a shame to waste that experience. Rather there seems little downside in letting old-hands run.

Author: It does come from the existing text, and I saw it as the other end of a time bound - the first clause says "you must have been doing stuff for at least this long" and this one says "and also relatively recently".

I do agree it may be a bit too harsh, though. I have some friends on Apache PMCs who had some different advice about avoiding major conflicts of interest so I might see if I can replace it with wording for that.

Reviewer: Having this conversation it occurs to me that we can frame the issue as two interrelated problems, maybe: 

1. This time-limit serves to rule out candidates because they rotated out for a period. 
2. The narrow tech focus meant it's only _code contributors_ who are allowed. 

Changing the bullets fixes 2. Dropping this would fix 1. 

(Or something like that 😜)

🎁

Author: @carltongibson I have tried to rework this to keep some of the spirit of the original without a time bound - thoughts?

Additionally, the following section is added to the definition of
the Steering Council to make its role clearer:

  The Council's goal is twofold - to safeguard big decisions that affect
  Django projects at a fundamental level, and to help shepherd the project's
  future direction.

>   While the Council should not to define this direction entirely by itself,

Reviewer 4: Either a verb is missing, or:

```suggestion
  While the Council should not define this direction entirely by itself,
```

  it should be the catalyst within the community for doing so - as such, it is
  expected for Council members to actively participate in engaging with the
  community, canvassing for ideas about big new features or directions to take
  the framework, and reporting back to the community and the DSF Board on these
  ideas and if the Council believes they should be followed.

Motivation
==========

While the model outlined by DEP 10 is still valid, the last few Technical
Boards have not done the entire role in the way that was originally
envisioned - instead, mostly acting as a last-hurdle approval on DEPs and
appointments to other positions.

In recent elections we have also faced a lack of candidates, culminating in the
most recent one, where we had only five candidates for five positions. It is
not just Django specific - talking to friends who help with other Open Source
projects in a variety of languages, and indeed to non-profits at large, this
appears to be a wider trend.

In order to get closer to the initial intention, we want to make the Board more

> active, but it is apparent that to do this we must staff it more sustainably.

Reviewer 4: ```suggestion
active, but it is apparent that, to do this, we must staff it more sustainably.
```
(Makes the flow a tiny bit more parsable)

As such, this DEP does two things:

* Makes the Board's role more clear by renaming it (Steering Council implies
  the job of "working out what features and direction to go" more than
  Technical Board does) and by specifically adding language highlighting the
  more active role expected of its members.

* Increases the number of people eligible for candidacy on the Council in
  order for us to sustainably staff it for the foreseeable future, while not

>   making the rules so loose that they risk hostile or populist takeovers.

Reviewer 4: ```suggestion
  making the rules so loose that it risks hostile or populist takeovers.
```
"They" would refer to "the rules" ? I'd say the entity at risk is "the Council" ?

>   making the rules so loose that it risks hostile or populist takeovers.

Reviewer 5: ```suggestion
  making the rules so loose that it risks hostile takeovers.
```
You might consider chopping this word, as it can have either positive or negative connotations depending on the subject matter and on political views--which could muddle the intended negative connotation.

It refrains from making any further changes under the belief that good progress
is incremental - take a model, refine it slightly, and then if we continue to
see problems, we should react to those. If the path suggested by this DEP does
not improve matters by the end of the 5.x release series, it is the author's
belief that further changes may be necessary.

Rationale
=========

> While we could try and redo our governance process far more drastically, I
> believe that the model outlined in DEP 10 is fundamentally sound and merely

Reviewer 4: ```suggestion
While we could try and redo our governance process far more drastically, the
author believes that the model outlined in DEP 10 is fundamentally sound and merely
```
(Same)

needs a few tweaks as the demographics of who is willing to step up and help
steer Django has changed over the years.

> We also could do nothing, but I think that would be foolish in the face of the

Reviewer 4: ```suggestion
We also could do nothing, but the author thinks that would be foolish in the face of the
```
(Completely optionnal, just it's not always obvious who "I" is on a document like this, especially once it will be accepted)

current approach (which is not what was outlined in DEP 10 anyway) not working.

> Change is needed, and I believe this is the right line between iterative

Reviewer 4: ```suggestion
Change is needed, and the author believes this is the right line between iterative
```
(missed one)

  future direction.

  While the Council SHOULD NOT define this direction entirely by itself,
  it should be the catalyst within the community for doing so - as such, it is
  expected for Council members to actively participate in engaging with the
  community, canvassing for ideas about big new features or directions to take
  the framework, and reporting back to the community and the DSF Board on these
  ideas and if the Council believes they should be followed.

  Any Django Software Foundation individual member may make a public statement
  of no-confidence in the Steering Council by identifying a material breach of
  their duties as defined in DEP 10 and DEP 12. Upon seconding by another
  individual member of the DSF the DSF Board SHALL no later than the next
  scheduled board meeting evaluate the merits of the statement of
  no-confidence.

  If the statement is found to be accurate and correct the Board shall inform
  the Steering Council of the breach and provide 2 weeks to rectify said
  breach. If the Steering Council fails to rectify the breach in the time
  allotted, a new Steering Council election SHALL be triggered. Current members

>   of the Steering Council are permitted to run in the new election.

Reviewer 4: ```suggestion
  of the Steering Council MAY run in the new election.
```

  * Running Django-related events or user groups

* A history of engagement with the direction and future of Django.
  This does not need to be recent, but candidates who have not engaged in the
  past three years must still demonstrate an understanding of Django's changes

>   and direction within those three years.

Reviewer: Yeah, nice. Maybe folks can suggest tweaks, but I think that covers both the original motivation (or what I read that as, at least) **and** allows people to cycle back in, which is the observation that's been made about the current wording. 

👍

Reviewer: Thanks Andrew!
